tools/kata-agent-ctl/src/client.rs

```diff
@@ -0,0 +1,1193 @@
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Description: Client side of ttRPC comms
+
+use crate::utils;
+use nix::sys::socket::{connect, socket, AddressFamily, SockAddr, SockFlag, SockType};
+use protocols::agent::*;
+use protocols::agent_ttrpc::*;
+use protocols::health::*;
+use protocols::health_ttrpc::*;
+use slog::{debug, info};
+use std::io;
+use std::io::Write; // XXX: for flush()
+use std::os::unix::io::RawFd;
+use std::thread::sleep;
+use std::time::Duration;
+
+use crate::types::{Config, Options, Result};
+
+// Agent command handler type
+//
+// Notes:
+//
+// - 'cmdline' is the command line (command name and optional space separate
+//   arguments).
+// - 'options' can be read and written to, allowing commands to pass state to
+//   each other via well-known option names.
+type AgentCmdFp = fn(
+    cfg: &Config,
+    client: &AgentServiceClient,
+    health: &HealthClient,
+    options: &mut Options,
+    args: &str,
+) -> Result<()>;
+
+// Builtin command handler type
+type BuiltinCmdFp = fn(cfg: &Config, options: &mut Options, args: &str) -> (Result<()>, bool);
```

Comment:
I think this function could return `Result<bool>`, this would simplify things.

Reply:
yep - I guess this makes sense - will fix tomorrow...